crash in widget.(*RichText).lineSizeToColumn

### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

We used the Fuzz engine to modify some Test(TestEntry_EraseSelection) data, and then the following crash appeared. Triggered when the value of the exported member (e.CursorColumn) is negative.

### How to reproduce

The simplified test code is in **Example code**

### Screenshots

CrashLog
```
panic: runtime error: slice bounds out of range [:-1] [recovered]
        panic: runtime error: slice bounds out of range [:-1]

goroutine 63 [running]:
testing.tRunner.func1.2({0x87e8c0, 0xc000024138})
        /home/zjx/.local/go/src/testing/testing.go:1211 +0x24e
testing.tRunner.func1()
        /home/zjx/.local/go/src/testing/testing.go:1214 +0x218
panic({0x87e8c0, 0xc000024138})
        /home/zjx/.local/go/src/runtime/panic.go:1038 +0x215
fyne.io/fyne/v2/widget.(*RichText).lineSizeToColumn(0xc0001c0580, 0xffffffffffffffff, 0x60)
        /home/zjx/workspace/gowork/src/purelib/crashConfirm/fyne/widget/richtext.go:253 +0x33a
fyne.io/fyne/v2/widget.(*entryContentRenderer).buildSelection.func1(0x863a80, 0x0)
        /home/zjx/workspace/gowork/src/purelib/crashConfirm/fyne/widget/entry.go:1569 +0x2b
fyne.io/fyne/v2/widget.(*entryContentRenderer).buildSelection(0xc00012c1e0)
        /home/zjx/workspace/gowork/src/purelib/crashConfirm/fyne/widget/entry.go:1619 +0x512
fyne.io/fyne/v2/widget.(*entryContentRenderer).moveCursor(0xc00012c1e0)
        /home/zjx/workspace/gowork/src/purelib/crashConfirm/fyne/widget/entry.go:1665 +0x25
fyne.io/fyne/v2/widget.(*entryContentRenderer).Refresh(0xc00012c1e0)
        /home/zjx/workspace/gowork/src/purelib/crashConfirm/fyne/widget/entry.go:1533 +0x1b4
fyne.io/fyne/v2/widget.(*entryRenderer).Refresh(0xc0000207c0)
        /home/zjx/workspace/gowork/src/purelib/crashConfirm/fyne/widget/entry.go:1407 +0x482
fyne.io/fyne/v2/widget.(*BaseWidget).Refresh(0xc000213de0)
        /home/zjx/workspace/gowork/src/purelib/crashConfirm/fyne/widget/widget.go:138 +0x2e
fyne.io/fyne/v2/widget.(*BaseWidget).setFieldsAndRefresh(0xc00000c1e0, 0xc000213de8)
        /home/zjx/workspace/gowork/src/purelib/crashConfirm/fyne/widget/widget.go:152 +0x71
fyne.io/fyne/v2/widget.(*Entry).updateText(0xc00000c1e0, {0xc000024120, 0x17})
        /home/zjx/workspace/gowork/src/purelib/crashConfirm/fyne/widget/entry.go:1172 +0x7c
fyne.io/fyne/v2/widget.(*Entry).TypedKey(0xc00000c1e0, 0xc00000e390)
        /home/zjx/workspace/gowork/src/purelib/crashConfirm/fyne/widget/entry.go:648 +0x605
fyne.io/fyne/v2/widget.TestEntry_EraseSelection.func1(0x855d00)
        /home/zjx/workspace/gowork/src/purelib/crashConfirm/fyne/widget/entry_internal_test.go:242 +0x99
fyne.io/fyne/v2/widget.TestEntry_EraseSelection.func2(0xc00000e390)
        /home/zjx/workspace/gowork/src/purelib/crashConfirm/fyne/widget/entry_internal_test.go:245 +0x2f
fyne.io/fyne/v2/widget.TestEntry_EraseSelection(0x0)
        /home/zjx/workspace/gowork/src/purelib/crashConfirm/fyne/widget/entry_internal_test.go:249 +0x163
```

### Example code

```go
//part of the Test Code
	e := NewMultiLineEntry()
	e.SetText("00000000000000000000000")
	e.CursorRow = 0
	e.CursorColumn = -1

	var keyDown = func(key *fyne.KeyEvent) {
		e.KeyDown(key)
		e.TypedKey(key)
	}
	var keyPress = func(key *fyne.KeyEvent) {
		keyDown(key)
		e.KeyUp(key)
	}
	keyDown(&fyne.KeyEvent{Name: desktop.KeyShiftLeft})
	keyPress(&fyne.KeyEvent{Name: fyne.KeyRight})
```

### Fyne version

commit 638ae24e4bebfc0b6dbaabf8db5807eeabfafc61 (HEAD -> master, origin/master, origin/HEAD)

### Go compiler version

1.17.8

### Operating system

Linux

### Operating system version

OS: Linux r920 4.15.0-159-generic #167-Ubuntu

### Additional Information

_No response_

--- a/widget/entry.go
+++ b/widget/entry.go
@@ -438,6 +438,9 @@ func (e *Entry) SelectedText() string {
 	}
 
 	start, stop := e.selection()
+	if start == stop {
+		return ""
+	}
 	e.propertyLock.RLock()
 	defer e.propertyLock.RUnlock()
 	r := ([]rune)(e.textProvider().String())

--- a/widget/entry_test.go
+++ b/widget/entry_test.go
@@ -353,6 +353,11 @@ func TestEntry_EmptySelection(t *testing.T) {
 	typeKeys(entry, fyne.KeyRight, keyShiftLeftDown, fyne.KeyRight, fyne.KeyLeft, keyShiftLeftUp)
 	assert.Equal(t, "", entry.SelectedText())
 	assert.Equal(t, 1, entry.CursorColumn)
+
+	// manually setting to empty selection
+	typeKeys(entry, keyShiftLeftDown, fyne.KeyRight)
+	entry.CursorColumn = 1
+	assert.Equal(t, "", entry.SelectedText())
 }
 
 func TestEntry_Focus(t *testing.T) {

--- a/widget/richtext.go
+++ b/widget/richtext.go
@@ -252,6 +252,12 @@ func (t *RichText) len() int {
 
 // lineSizeToColumn returns the rendered size for the line specified by row up to the col position
 func (t *RichText) lineSizeToColumn(col, row int) fyne.Size {
+	if row < 0 {
+		row = 0
+	}
+	if col < 0 {
+		col = 0
+	}
 	bound := t.rowBoundary(row)
 	total := fyne.NewSize(0, 0)
 	counted := 0

--- a/widget/richtext_test.go
+++ b/widget/richtext_test.go
@@ -370,6 +370,9 @@ func TestTextProvider_LineSizeToColumn(t *testing.T) {
 	fullSize := provider.lineSizeToColumn(4, 0)
 	assert.Equal(t, fullSize, provider.lineSizeToColumn(10, 0))
 	assert.Greater(t, fullSize.Width, provider.lineSizeToColumn(2, 0).Width)
+
+	out := provider.lineSizeToColumn(-1, -1)
+	assert.Equal(t, out, provider.lineSizeToColumn(0, 0))
 }
 
 func TestText_splitLines(t *testing.T) {

